pkg/asset/manifests/azure/cluster.go

```diff
@@ -80,21 +84,145 @@ func GenerateClusterAssets(installConfig *installconfig.InstallConfig, clusterID
+										SourcePorts:      ptr.To("*"),
+										DestinationPorts: ptr.To("22"),
+										Source:           ptr.To("*"),
+										Destination:      ptr.To("*"),
+										Action:           capz.SecurityRuleActionAllow,
+									},
+									{
+										Name:             "http",
+										Protocol:         capz.SecurityGroupProtocolTCP,
+										Direction:        capz.SecurityRuleDirectionInbound,
+										Priority:         104,
+										SourcePorts:      ptr.To("*"),
+										DestinationPorts: ptr.To("80"),
+										Source:           ptr.To("*"),
+										Destination:      ptr.To("*"),
+										Action:           capz.SecurityRuleActionAllow,
+									},
+									{
+										Name:             "https",
+										Protocol:         capz.SecurityGroupProtocolTCP,
+										Direction:        capz.SecurityRuleDirectionInbound,
+										Priority:         105,
+										SourcePorts:      ptr.To("*"),
+										DestinationPorts: ptr.To("443"),
+										Source:           ptr.To("*"),
+										Destination:      ptr.To("*"),
+										Action:           capz.SecurityRuleActionAllow,
+									},
+								},
+							},
+						},
 					},
 					{
 						SubnetClassSpec: capz.SubnetClassSpec{
-							Name: "worker-subnet",
+							Name: computeSubnet,
 							Role: capz.SubnetNode,
 							CIDRBlocks: []string{
 								subnets[1].String(),
 							},
 						},
+						SecurityGroup: capz.SecurityGroup{
+							Name: fmt.Sprintf("%s-worker-nsg", clusterID.InfraID),
+							SecurityGroupClass: capz.SecurityGroupClass{
+								SecurityRules: []capz.SecurityRule{
+									{
+										Name:             "apiserver_in",
+										Protocol:         capz.SecurityGroupProtocolTCP,
+										Direction:        capz.SecurityRuleDirectionInbound,
+										Priority:         101,
+										SourcePorts:      ptr.To("*"),
+										DestinationPorts: ptr.To("6443"),
+										Source:           ptr.To("*"),
+										Destination:      ptr.To("*"),
+										Action:           capz.SecurityRuleActionAllow,
+									},
+									{
+										Name:             "ignition",
+										Protocol:         capz.SecurityGroupProtocolTCP,
+										Direction:        capz.SecurityRuleDirectionInbound,
+										Priority:         102,
+										SourcePorts:      ptr.To("*"),
+										DestinationPorts: ptr.To("22623"),
+										Source:           ptr.To("*"),
+										Destination:      ptr.To("*"),
+										Action:           capz.SecurityRuleActionAllow,
+									},
+									{
+										Name:             "ssh",
+										Protocol:         capz.SecurityGroupProtocolTCP,
+										Direction:        capz.SecurityRuleDirectionInbound,
+										Priority:         103,
+										SourcePorts:      ptr.To("*"),
+										DestinationPorts: ptr.To("22"),
+										Source:           ptr.To("*"),
+										Destination:      ptr.To("*"),
+										Action:           capz.SecurityRuleActionAllow,
+									},
```

Comment:
Ultimately we will want to remove this rule along with the bootstrap node.

Reply:
Removed.